File: example/modules/ar-detectors/android/src/main/java/expo/modules/ardetectors/ArDetectorsModule.kt
package expo.modules.ardetectors

import expo.modules.ar.ExpoArDetectorRegistry
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// Registers the example ML Kit processor into expo-ar's registry at app startup. The registry name
// ("objects") must match the <ExpoArView detectionModel="objects" /> prop in App.tsx. No JS API.
class ArDetectorsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ArDetectors")

    OnCreate {
      ExpoArDetectorRegistry.register("objects", MlKitObjectProcessor())
    }
  }
}

File: example/modules/ar-detectors/ios/ArDetectorsModule.swift
import ExpoAr
import ExpoModulesCore

// Registers the example Vision processor into expo-ar's registry at app startup. The registry name
// ("objects") must match the <ExpoArView detectionModel="objects" /> prop in App.tsx. No JS API.
public class ArDetectorsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ArDetectors")

    OnCreate {
      ExpoArDetectorRegistry.register(name: "objects", processor: VisionObjectProcessor())
    }
  }
}
